app/templates/components/main-gist.hbs

```diff
@@ -1,51 +1,55 @@
-{{#if this.fileTreeShown}}
-  {{gist-header model=model
-                session=session
-                unsaved=unsaved
-                isGistSaving=isGistSaving
-                activeEditorCol=activeEditorCol
-                activeFile=activeFile
-                isRevision=isRevision
-                selectedKeyMap=settings.keyMap
-                fileTreeShown=fileTreeShown
-                titleChanged=(action "titleChanged")
-                addFile=(action "addFile")
-                addHelper=(action "addHelper")
-                addComponent=(action "addComponent")
-                addUnitTestFile=(action "addUnitTestFile")
-                addIntegrationTestFile=(action "addIntegrationTestFile")
-                addAcceptanceTestFile=(action "addAcceptanceTestFile")
-                renameFile=(action "renameFile")
-                removeFile=(action "removeFile")
-                setEditorKeyMap=(action "setEditorKeyMap")
-                versionSelected=(action "versionSelected")
-                openFile=(action "openFile")
+<SidebarLayout
+  @session={{@session}}
+  @isSidebarOpen={{@isSidebarOpen}}
+  @onSidebarUpdated={{@onSidebarUpdated}}
+  @sidebarComponent={{component "gist-header"
```

Comment:
Instead of passing in a component, can we yield a template that calls it?

Reply:
Not really, unless we use a wormhole. Yieldable blocks will help once they land.